src/benchq/problem_ingestion/hamiltonian_generation.py

```diff
@@ -83,3 +130,87 @@ def generate_1d_heisenberg_hamiltonian(N):
     for term in pauli_sum.terms:
         term.coefficient = term.coefficient.real
     return pauli_sum
+
+
+def generate_kitaev_hamiltonian(lattice_size, weight_prob=1):
+    """
+    \sum_{Z edges} j_{Z edge} \sigma_z \sigma_z + \sum_{x edges} j_{X edge} \sigma_x \sigma_x +
+    \sum_{Y edges} j_{Y edge} \sigma_y \sigma_y
+    """
+
+    def assign_hexagon_labels(g):
+        for n1, n2 in g.edges:
+            #start by making sure that the edges are ordered correctly
+            r1,c1 = n1
+            r2,c2 = n2
+            if r2 - r1 < 0 or c2 - c1 < 0:
+                swap_r2 = r1
+                swap_c2 = c1
+                r1 = r2
+                c1 = c2
+                r2 = swap_r2
+                c2 = swap_c2
+            
+            #now that they are ordered correctly, we can assign labels
+            label = ''
+            if c1 == c2:
+                label = 'Z'
+            elif (((r1 % 2) + (c1 % 2)) % 2 == 0):#apparently you can differentiate X and Y labels based off nx's node label parity.  Huh.
+                label = 'Y'
+            else:
+                label = 'X'
+            
+            g[n1][n2]['label'] = label
+
+    def nx_kitaev_terms(g, p):
+        H = []
+        n = len(g.nodes)
+        for (n1,n2,d) in g.edges(data=True):
+            label = d['label']
+            weight = 1 if random.random() < p else -1
+            string = n*'I' 
+            for i in range(len(g)):
+                if i == n1 or i == n2:
+                    string = string[:i] + label + string[i+1:]
+                else:
+                    pass
+            H.append((string,weight))
+        return H
```

Comment:
Why are these functions defined inside this function? I dont think they use the outter function's arguments or anything like that?

Reply:
@AthenaCaesura  I put them inside since I don't imagine they would be used by any other function apart from. Is this bad form ?

Reply:
No, it just looks a little strange. Usually you would do something like this if you really **really** want to hide these functions from users. If you don't want users to use these functions typically, I would just make them "pseudoprivate" by putting underscores below them. (sorry it took me so long to respond here).